polynote-frontend/polynote/ui.js

```diff
@@ -1964,17 +2005,33 @@ export class MainUI extends EventTarget {
     }
 
     createNotebook(evt) {
-        const notebookPath = prompt("Enter the name of the new notebook (no need for an extension)");
-        if (notebookPath) {
-            const handler = this.socket.addMessageListener(messages.CreateNotebook, (actualPath) => {
-                this.socket.removeMessageListener(handler);
-                this.browseUI.addItem(actualPath);
-                this.loadNotebook(actualPath);
-            });
+        const handler = this.socket.addMessageListener(messages.CreateNotebook, (actualPath) => {
+            this.socket.removeMessageListener(handler);
+            this.browseUI.addItem(actualPath);
+            this.loadNotebook(actualPath);
+        });
 
-            this.socket.send(new messages.CreateNotebook(notebookPath))
+        if (evt.detail && evt.detail.name) { // the evt has all we need
+            this.socket.send(new messages.CreateNotebook(evt.detail.name, Either.right(evt.detail.content)));
+        } else {
+            const notebookPath = prompt("Enter the name of the new notebook (no need for an extension), or the full URL of another Polynote instance.");
+
+            if (notebookPath && notebookPath.startsWith("http")) {
+                const nbFile = decodeURI(notebookPath.split("/").pop());
+                const targetPath = notebookPath + "?download=true";
+                this.socket.send(new messages.CreateNotebook(nbFile, Either.left(targetPath)));
+            } else if (notebookPath) {
+                this.socket.send(new messages.CreateNotebook(notebookPath))
+            }
         }
     }
+
+    static browserDownload(path, filename) {
```

Comment:
Should we just set the `Content-Disposition` header on the server side? I think that would make this unnecessary?

Reply:
On second thought, if this way ain't broke don't fix it